services/checks/checks.go

```diff
@@ -416,6 +422,82 @@ func (s *Service) EnableChecks(checkNames []string) error {
 	return nil
 }
 
+// ChangeInterval changes a check's interval to the value received from the UI.
+func (s *Service) ChangeInterval(checkName string, interval managementpb.SecurityCheckInterval) error {
+	for _, c := range s.GetAllChecks() {
+		if c.Name == checkName {
+			switch interval {
+			case managementpb.SecurityCheckInterval_STANDARD:
+				c.Interval = check.Standard
+			case managementpb.SecurityCheckInterval_FREQUENT:
+				c.Interval = check.Frequent
+			case managementpb.SecurityCheckInterval_RARE:
+				c.Interval = check.Rare
+			default:
+				return errors.New("invalid security check interval")
+			}
+
+			s.updateCheckInterval(c)
+
+			// since we re-run checks at regular intervals using a call
+			// to s.StartChecks which in turn calls s.collectChecks
+			// to load/download checks, we must record any changes
+			// to check intervals in this map so that they can be re-applied
+			// once the checks have been re-loaded on restarts.
+			s.cim.Lock()
+			defer s.cim.Unlock()
+			s.checksIntervalMap[c.Name] = c.Interval
+			break
+		}
+	}
+	return nil
+}
+
+func (s *Service) updateCheckInterval(newCheck check.Check) {
```

Comment:
This function updates the whole check, not just the interval. So it should have a different name or different signature.

Reply:
renamed